src/components/notes/add.jsx

```diff
@@ -1,65 +1,42 @@
-import React, { Component } from 'react'
+import React, { useState, useCallback } from 'react'
 
 import { withClient } from 'cozy-client'
 
-import { withRouter } from 'react-router-dom'
-
 import Button from 'cozy-ui/react/Button'
 import { translate } from 'cozy-ui/react/I18n'
+import BarButton from 'cozy-ui/react/BarButton'
 
-import { schemaOrdered } from 'lib/collab/schema'
-import { generateReturnUrlToNotesIndex } from 'lib/utils'
-
-class Add extends Component {
-  constructor(props, context) {
-    super(props, context)
-    // initial component state
-    this.state = {
-      isWorking: false
-    }
-  }
-
-  handleClick = async () => {
-    this.setState(() => {
-      true
-    })
-    const { client: cozyClient } = this.props
-    const { data: doc } = await cozyClient
-      .getStackClient()
-      .fetchJSON('POST', '/notes', {
-        data: {
-          type: 'io.cozy.notes.documents',
-          attributes: {
-            title: '',
-            schema: schemaOrdered
-          }
-        }
-      })
-    this.setState(() => {
-      false
-    })
+import { createNoteDocument, generateReturnUrlToNotesIndex } from 'lib/utils'
 
+const Add = ({ t, className, client }) => {
+  const [isWorking, setIsWorking] = useState(false)
+  const handleClick = useCallback(async () => {
+    setIsWorking(true)
+    const { data: doc } = await createNoteDocument(client)
     window.location.href = generateReturnUrlToNotesIndex(doc)
-  }
-
-  render() {
-    const { isWorking } = this.state
-    const { t, className } = this.props
-    return (
-      <div>
-        <Button
-          onClick={this.handleClick}
-          type="submit"
-          busy={isWorking}
-          icon="plus"
-          label={t('Notes.Add.add_note')}
-          extension="narrow"
-          className={className}
-        />
-      </div>
-    )
-  }
+  }, [])
+
+  return (
+    <Button
+      onClick={handleClick}
+      type="submit"
+      busy={isWorking}
+      icon="plus"
+      label={t('Notes.Add.add_note')}
+      extension="narrow"
+      className={className}
+    />
+  )
 }
 
-// get mutations from the client to use createDocument
-export default translate()(withClient(withRouter(Add)))
+export const AddMobile = withClient(({ client }) => (
+  <BarButton
+    onClick={async () => {
+      const { data: doc } = await createNoteDocument(client)
+      window.location.href = generateReturnUrlToNotesIndex(doc)
+    }}
```

Comment:
We should wrap this in a `useCallback`, like in the other Add button.

Reply:
We can yep but I'm curious of the gain you see to do so.